sympy/solvers/ode/riccati.py

```diff
@@ -0,0 +1,506 @@
+Find `b_0, b_1` and `b_2`. If `b_2 = 0` or no such functions exist, raise
+an error
+
+Step 1 : Transform the equation to its normal form as explained in the
+theory section.
+
+Step 2 : Initialize 2 empty set of solutions, ``sol`` and ``presol``.
+
+Step 3 : If `a(x) = 0`, append `\frac{1}/{(x - C1)}` to ``presol``.
+
+Step 4 : If `a(x)` is a rational non-zero number, append `\pm \sqrt{a}`
+to ``presol``.
+
+Step 5 : Find the poles and their multiplicities of `a(x)` using
+``find_poles``. Let the number of poles be `n`. Also find the
+valuation of `a(x)` at `\infty` using ``val_at_inf``.
+
+Step 6 : Find `n` c-vectors (one for each pole) and 1 d-vector using
+``construct_c`` and ``construct_d``. Now, determine all the ``2**(n + 1)``
+combinations of choosing between 2 choices for each of the `n` c-vectors
+and 1 d-vector.
+
+For each of these above combinations, do
+
+Step 8 : Compute `m` in ``compute_degree``
+
+Step 9 : In ``compute_degree``, compute ybar as well. If ybar contains any
+term like `\frac{c_{ij}}{(x - oo)^j}` where `c_{ij}` is non-zero, discard
+this solution.
+
+Step 10 : If `m` is a non-negative integer -
+
+Step 11: Find a polynomial solution of degree `m` for the auxiliary equation.
+
+There are 2 cases possible -
+
+    a. `m` is a non-negative integer: We can solve for the coefficients
+    in `p(x)` using Undetermined Coefficients.
+
+    b. `m` is an expression: In this case, we may not be able to determine
+    if it is possible for `m` to be a non-negative integer. However, the
+    solution to the auxiliary equation always seems to be x**m + C1 (where
+    C1 is an arbitrary constant). This has been confirmed by testing many
+    examples. NOTE that this is NOT mentioned in the thesis and is an
+    assumption made by me. If any failing test cases are found, this case
+    must be removed.
+
+Step 12 : For each `p(x)` that exists, append `ybar + \frac{p'(x)}{p(x)}`
+to ``presol``.
+
+Step 13 : For each solution in ``presol``, apply an inverse transformation
+and append them to ``sol``, so that the solutions of the original equation
+are found using the solutions of the equation in its normal form.
+"""
+
+from sympy.core import S
+from sympy.core.numbers import oo
+from sympy.core.relational import Eq
+from sympy.core.symbol import symbols, Dummy
+from sympy.functions import sqrt
+from sympy.functions.elementary.complexes import sign
+from sympy.polys.domains import ZZ
+from sympy.polys.polytools import Poly
+from sympy.polys.polyroots import roots
+from sympy.solvers.solveset import linsolve
+
+def riccati_normal(w, x, b1, b2):
+    # y(x) = -b2(x)*w(x) - b2'(x)/(2*b2(x)) - b1(x)/2
+    return -b2*w - b2.diff(x)/(2*b2) - b1/2
+
+
+def riccati_inverse_normal(y, x, b1, b2, bp=None):
+    if bp is None:
+        bp = -b2.diff(x)/(2*b2**2) - b1/(2*b2)
+    # w(x) = -y(x)/b2(x) - b2'(x)/(2*b2(x)^2) - b1(x)/(2*b2(x))
+    return -y/b2 + bp
+
+
+def linsolve_dict(eq, syms, var):
+    sol = linsolve(eq, syms, var)
```

Comment:
If `syms` is a list then `linsolve` will ignore the `var` argument. Why is that passed?

Reply:
`syms` is a list of undetermined coefficient symbols, while `var` is `x`.

Reply:
I think that the var argument is just being ignored so it shouldn't be passed to `linsolve`.

Reply:
Yes, it is being ignored. Removed it.